main.cpp

```diff
@@ -1,7 +1,26 @@
 #include "Character.h"
 #include "Player.h"
 #include <iostream>
-#include <fstream>
+
+void Fight(Player player, Character character) {
```

Comment:
Itt is referenciaként kéne átadni a harcosokat. És ne csak Player-Char harc lehessen, hanem Player-Player és Char-Char harc is.

Reply:
@oliverosz A referenciát értem, de nem maradhatna így hogy Player küzd Characterrrel? Hiszen ez volt a feladat és miért kellene két azonos fajtának is küzdenie egymással? Ugye most az egész logika erre épül, akkor megint át kellene írni az szinte az egész kódót, legalábbis így gondolom. Vagy légyszíves mond el ha mégis így szeretnéd hogyan lehetne ezt kivitelezni, szeretném a segítségedet kérni. Esetleg ha maradhatna így, vagy ha megírod hogyan kellene módosítanom, azt előre is köszönöm!

Reply:
Úgy, hogy Character& legyen mindkét paraméter. Pont az a lényege a polimorfizmusnak, hogy ehhez nem kell változtatni a logikán, mert a Fightnak nem kell tudnia, hogy pontosan milyen típusú harcosokat kap. Csak ehhez virtuálisnak kell lennie a metódusnak, ahogy ott jeleztem.

Reply:
@oliverosz Így már értem akkor, bocsi hogy ez nem tudott az eszembe jutni és köszönöm a segítséget!